frontend/src/tests/components/commons/ProgressPane.test.js

```diff
@@ -11,29 +13,54 @@ const PROGRESS = [
 ];
 
 it("renders when first node selected", () => {
-  testCore(0);
+  generateAndTestCore(0);
 });
 
 it("renders when second node selected", () => {
-  testCore(1);
+  generateAndTestCore(1);
 });
 
 it("renders when third node selected", () => {
-  testCore(2);
+  generateAndTestCore(2);
 });
 
 it("renders when fourth node selected", () => {
-  testCore(3);
+  generateAndTestCore(3);
 });
 
-function testCore(selected) {
+function generateAndTestCore(selected) {
   const wrapper = mount(<ProgressPane progressSpecs={PROGRESS} currentNode={selected} />);
-  const node1 = <ProgressNode id={0} text={"node 1"} current={selected} />;
-  const node2 = <ProgressNode id={1} text={"node 2"} current={selected} />;
-  const node3 = <ProgressNode id={2} text={"node 3"} current={selected} />;
-  const node4 = <ProgressNode id={3} text={"node 4"} current={selected} />;
-  expect(wrapper.contains(node1)).toEqual(true);
-  expect(wrapper.contains(node2)).toEqual(true);
-  expect(wrapper.contains(node3)).toEqual(true);
-  expect(wrapper.contains(node4)).toEqual(true);
+  testCore(selected, wrapper);
+}
+
+function testCore(selected, wrapper) {
```

Comment:
This is cool! I think it could use some comments.

Reply:
comments added in 03b02b9